src/core/math_func.hpp

```diff
@@ -83,6 +83,29 @@ static inline T Clamp(const T a, const T min, const T max)
 	return a;
 }
 
+/**
+ * Clamp a value between an interval.
+ *
+ * This function returns a value which is between the given interval of
+ * min and max. If the given value is in this interval the value itself
+ * is returned otherwise the border of the interval is returned, according
+ * which side of the interval was 'left'.
+ *
+ * @note If the min value is greater than the max, return value is the average of the min and max.
+ * @param a The value to clamp/truncate.
+ * @param min The minimum of the interval.
+ * @param max the maximum of the interval.
+ * @returns A value between min and max which is closest to a.
+ */
+template <typename T>
+static inline T SoftClamp(const T a, const T min, const T max)
+{
+	if (min > max) return (min + max) / 2;
```

Comment:
This is broken: https://www.youtube.com/watch?v=sBtAGxBh-XI&t=278s but C++20 to the rescue!

Reply:
These comments always puzzle me .. do you want us to address the problem? Or was it just meant as fun: hihi, strictly seen this can break! 

For our use-cases the current is ofc fine; but yeah .. let me know how you meant this comment, and we take it from there :)

Reply:
Right, so you added a unit-test that broke the CI, what this comment was referencing. Please, PLEASE, use a bit more words when you try to say something :) It really saves us a lot of time :P

Anyway, I fixed it .. bit rude to not fix it yourself, but okay :D :) (just kidding ofc; I am fine with fixing it)

Reply:
@JGRennison might be worth for you to know too, that `SoftClamp` could overflow and return invalid midpoints; I fixed it in this PR :)

Reply:
In my opinion a function that is as generic as this one is, should also work right in reasonably conceivable cases.

After all, you add it here to be reused in many other places and based on the comment you will not know that it is broken in some cases. To make matters weirder, with any type T that is smaller than `int` this function will return the expected value als the math gets promoted to `int` first.

At the location where you currently use it, it's indeed reasonable to consider it to not be a problem because one value is always 0. Though if this starts to get used throughot the code I rather this known issue solved, than to find out about it in some weird bug report/crash where you are wondering why `SoftClamp` returns "garbage" in some cases.

Reply:
After finally finding somewhere what the C++ midpoint implementation was:

```
constexpr Integer midpoint(Integer a, Integer b) noexcept {
  using U = make_unsigned_t<Integer>;
  return a>b ? a-(U(a)-b)/2 : a+(U(b)-a)/2;
}
```

It has a nice point: making the left side unsigned also solves the corner case of extremely large `min` vs extremely small `max`. So added that to the mix. And added another test, to actually notice that scenario :) So it should now be 100% bug-free. Or something.